resharper/resharper-unity/src/Yaml/Psi/Modules/UnityExternalFilesModuleProcessor.cs

```diff
@@ -70,6 +74,9 @@ public class UnityExternalFilesModuleProcessor : IChangeProvider, IUnityReferenc
             if (!mySolutionDirectory.IsAbsolute)
                 mySolutionDirectory = solution.SolutionDirectory.ToAbsolutePath(FileSystemUtil.GetCurrentDirectory());
 
+            unityYamlDisableStrategy.Run(mySolutionDirectory);
```

Comment:
I don't recall discussing moving it outside of `OnUnityProjectAdded`. We can easily do this as part of `OnUnityProjectAdded` - on first call, walk `Assets`, `Packages` and `ProjectSettings`, collect all data and then act on it - add the filenames to a list and create them after the walk is finished.

If we're concerned about performance in very large projects it seems really weird to do this twice, especially as part of a class constructor. It feels like it would nearly double the "Preparing Unity project" stage (it's doing different things with the data, and hopefully the file system cache would improve things, but it's still walking the solution structure twice).